include/usb/usb_device.h

```diff
@@ -96,6 +96,31 @@ struct usb_setup_packet {
  * @{
  */
 
+/**
+ * @brief USB device State Codes
+ *
+ * State Codes gives an overview to user application what is USB device state
+ * For more information refer to chapter 9 from USB 2.0 spec.
+ */
+enum usb_state {
+	/** USB device is attached to the USB, but is not powered */
+	USB_DEVICE_ATTACHED,
+	/** USB device is attached and powered, but has not been reset */
+	USB_DEVICE_POWERED,
+	/** USB device is attached, powered and has been reset, but has not been
+	 * assigned a unique address. Device responds at the default address
+	 */
+	USB_DEVICE_DEFAULT,
+	/** USB device has been assigned a unique address */
+	USB_DEVICE_ADDRESS,
+	/** USB device was chosen a configuration */
```

Comment:
Device can be resumed from suspend and end up in not only configured state. From Suspend You can either enter Default, Address or Configured State. I had a trouble describing Configured state in one sentence because as I refered to Table 9-1 from usb 2.0 spec (chapter 9)
_"Device is attached to the USB, powered, has been reset, has a unique address, is configured, and is not suspended. The host may now use the function provided by the device."_

Maybe it would be better something like:
_"USB device is configured and the host may use the function provided by the device"_ ?

Reply:
No, usual it is Configured->Suspend->Configured. Not at all something like Configured->Suspend->Address.

Reply:
I meant depending in which state the device was before after entering Suspended it can either end up in x = (Powered, Default, Address, Configured). It's possible to have those transitions:
x -> Suspended -> x